state-synchronizer/src/request_manager.rs

```diff
@@ -329,12 +349,12 @@ impl RequestManager {
             .upstream_config
             .get_upstream_preference(peer.raw_network_id())
             == Some(0);
-        if is_primary_upstream_peer && self.multicast_level != MIN_UPSTREAM_NETWORK_CT {
+        if is_primary_upstream_peer && self.multicast_level != PRIMARY_NETWORK_PREFERENCE {
             // if chunk from a primary upstream is successful, stop multicasting the request to failover networks
             info!(LogSchema::event_log(LogEntry::Multicast, LogEvent::Recover)
                 .old_multicast_level(self.multicast_level)
-                .new_multicast_level(MIN_UPSTREAM_NETWORK_CT));
-            self.multicast_level = MIN_UPSTREAM_NETWORK_CT;
+                .new_multicast_level(PRIMARY_NETWORK_PREFERENCE));
+            self.multicast_level = PRIMARY_NETWORK_PREFERENCE;
```

Comment:
wouldn't it be better to just say

```
if peer.level < self.multicast_level {
  // update to a lower level
}

Reply:
+1 to ^, also probably worth having a function to save the boilerplate of info + set + counter.